### PR TITLE
doc: add dual navigation buttons and cleanup some docstrings

### PR DESCRIPTION
**Pull request by AbeCoull**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -19,18 +19,23 @@
...
-master_doc = "index"
+root_doc = "index"
 
 autoclass_content = "both"
 autodoc_member_order = "bysource"
 default_role = "py:obj"
 
 html_theme = "sphinx_rtd_theme"
+html_theme_options = {
+    "prev_next_buttons_location": "both",
```
Comment (laurencap) [inline doc/conf.py:30]: When I look at the docs artifact from the build (https://amazon-braket-sdk-python--758.org.readthedocs.build/en/758/) I still only see the next button.

Comment (math411) [reply doc/conf.py:30]: Is this only occurring on the landing page? If so, that's expected. Navigating using the next button, I can then see the previous button appear.

Comment (laurencap) [reply doc/conf.py:31]: Oh, of course, that makes perfect sense! 🤦‍♀️

**Review by laurencap — APPROVED** [review]

File: doc/conf.py [hunk 2]
```diff
@@ -20,18 +20,23 @@
...
+}
 htmlhelp_basename = "{}doc".format(project)
 
 language = "en"
 
 napoleon_use_rtype = False
+napoleon_google_docstring = True
+napoleon_numpy_docstring = False
```
Comment (speller26) [inline doc/conf.py:39]: Out of curiosity, why is this explicitly necessary?

**Review by speller26 — APPROVED** [review]

File: doc/conf.py (hunk 2 above)
Comment (math411) [reply doc/conf.py:38]: `napoleon_numpy_docstring` defaults to true so I was explicitly setting these values within the conf.

**Review by laurencap — APPROVED** [review]

**Pull request merged**